functions/src/create-contract.ts

```diff
@@ -56,11 +56,11 @@ export const createContract = newEndpoint(['POST'], async (req, [user, _]) => {
 
   let min, max, initialProb
   if (outcomeType === 'NUMERIC') {
-    ;({ min, max } = validate(numericSchema, req.body))
+    ;({ min, max } = validate(numericSchema, req.body))  // leading ; intentional: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#assignment_separate_from_declaration_2
```

Comment:
I kinda dislike these comments.

For one, It's not strictly true. The semicolon is not actually needed here as it's the first line in a block. Prettier adds the semicolon anyways because it adds one in front of every leading `(` and `[`. [(See reasoning here)](https://github.com/prettier/prettier/issues/736#issuecomment-291934981) 

Since we use Prettier, we don't actually need to document weird syntactic style. The autoformatter or the CI linter will catch it if someone tries to change it.

I just don't think these comments actually make the code more understandable.

If this line were really hard to read, it would make more sense to just rewrite it than to leave a comment on why it's written this way.

Reply:
+1

Reply:
Thanks for the detailed explanation, I realized I had only half understood the reasoning.